### PR TITLE
Cocoapods version

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

Ensures the required version of cocoapods is installed and used.

#### Ticket Link

n/a

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information

n/a

#### Screenshots

n/a

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -114,7 +114,7 @@ commands:
       - run:
           name: Getting cocoapods dependencies
           working_directory: ios
-          command: pod install
+          command: bundle exec pod install
```
Comment (mkraft) [inline .circleci/config.yml]: I have not tested this change.

File: Makefile [hunk 2]
```diff
@@ -22,6 +22,9 @@ node_modules: package.json
 	@echo Getting Javascript dependencies
 	@npm install
 
+gems: 
+	@bundle install
```
Comment (enahum) [inline Makefile]: Can we make sure this only runs if needed?

**Comment by mkraft** [comment]
> Made it so we check for the specific version of cocoapods available, if no required version found we install the needed version.
> 
> Also reverted the changes on circleci/config as that is working as intended.

I don't understand this commit. It seems to be doing what bundler and the Gemfile were invented to do.

**Comment by enahum** [comment]
@mkraft just not running bundle install if there is no need to run it

**Comment by enahum** [comment]
Although I can clean it up a bit more and remove the duplicate lines

**Comment by mkraft** [comment]
@enahum It's not about the duplicate lines, I just don't see the point. `bundle install` is just going to check what is locally installed. It won't re-download the gem if it's not required. What if cocoapods changes it's API for displaying the version? Or what if a dependent gem somehow isn't installed?

**Comment by enahum** [comment]
> What if cocoapods changes it's API for displaying the version?

Doubtful and can be updated in that case.

> Or what if a dependent gem somehow isn't installed?
After manually uninstalled?

The above will also be true for node dependencies and the actual pods, so I see no difference, I’m ok reverting it back in the end the difference will be another couple of seconds when running it

**Comment by mkraft** [comment]
> > What if cocoapods changes it's API for displaying the version?
> 
> Doubtful and can be updated in that case.
> 
> > Or what if a dependent gem somehow isn't installed?
> > After manually uninstalled?
> 
> The above will also be true for node dependencies and the actual pods, so I see no difference, I’m ok reverting it back in the end the difference will be another couple of seconds when running it

You decide

> > What if cocoapods changes it's API for displaying the version?
> 
> Doubtful and can be updated in that case.
> 
> > Or what if a dependent gem somehow isn't installed?
> > After manually uninstalled?
> 
> The above will also be true for node dependencies and the actual pods, so I see no difference, I’m ok reverting it back in the end the difference will be another couple of seconds when running it

You can replicate my issue with this approach by uninstalling a dependency, say, `gem uninstall atomos`. Then run `pod --version`. It'll throw an error. However, if you instead run `bundle install` it'll add the missing dependency.

Anyhow, you decide whether it's worth the performance difference, I just wanted to voice my concern.

**Pull request merged**